src/redist/targets/GenerateBundledVersions.targets

```diff
@@ -324,6 +324,12 @@
       <WindowsDesktop31RuntimePackRids Include="@(WindowsDesktop30RuntimePackRids)" />
       <WindowsDesktop50RuntimePackRids Include="@(WindowsDesktop31RuntimePackRids);win-arm64" />
       <WindowsDesktopRuntimePackRids Include="@(WindowsDesktop50RuntimePackRids)" />
+
+      <_KnownRuntimeIdentiferPlatforms Include="any;aot;freebsd;illumos;solaris;unix" />
+      <_ExcludedKnownRuntimeIdentiferPlatforms Include="rhel.6;tizen.4.0.0;tizen.5.0.0" Condition="'$(DotNetBuildFromSource)' != 'true'" />
+      <_ExcludedKnownRuntimeIdentiferPlatforms Include="rhel.6" Condition="'$(DotNetBuildFromSource)' == 'true' and !$(ProductMonikerRid.StartsWith('rhel.6-'))" />
+      <_ExcludedKnownRuntimeIdentiferPlatforms Include="tizen.4.0.0" Condition="'$(DotNetBuildFromSource)' == 'true' and !$(ProductMonikerRid.StartsWith('tizen.4.0.0-'))" />
+      <_ExcludedKnownRuntimeIdentiferPlatforms Include="tizen.5.0.0" Condition="'$(DotNetBuildFromSource)' == 'true' and !$(ProductMonikerRid.StartsWith('tizen.5.0.0-'))" />
```

Comment:
I don't really understand this part.  Is there a reason that the list can't be the same whether we're building from source or not?  And should the RID parsing in ProcessFrameworkReferences be removing the versions from the platforms so we don't have to do it here?

Reply:
I basically wanted to exclude those three non-portable RIDs (that we have runtime packs for) except if we are building from source and the non-portable RID (without the architecture) being used by source build matches one of them. We could also do it in the SDK based on `NETCoreSdkRuntimeIdentifier`, but I figured since we had all the information at generation time I'd add the logic in the generation and keep the actual SDK logic simpler.

I think we need to keep the versions. For example, if it is a non-portable source build for `rhel.8`, we want `rhel.8` to be considered as known (and not warn), but `rhel` or `rhel.9` should not be known (and should warn).